src/core/batch.rkt

```diff
@@ -145,6 +153,86 @@
       [_ node]))
   (unmunge reg))
 
+(define (batch-restore-index batch)
+  (make-hash (for/list ([node (in-vector (batch-nodes batch))]
+                        [n (in-naturals)])
+               (cons node n))))
+
+(define (egg-nodes->batch egg-nodes id->spec input-batch rename-dict)
+  (define out (make-mutable-batch))
+  (set-mutable-batch-index! out (batch-restore-index input-batch))
+
+  ; This fuction here is only because of cycles in loads:( Can not be imported from egg-herbie.rkt
```

Comment:
Yeah that's fine, reorganizing is definitely useful but it shouldn't hold us up.